python/tk_houdini/ui_generation.py

```diff
@@ -134,6 +134,98 @@ def _itemNode(self, parent, label, id):
         node.text = id
         return item
 
+class AppCommandsPanels(AppCommandsUI):
+
+    def __init__(self, engine, commands=None, panel_commands=None):
+        self._panel_commands = panel_commands
+        super(AppCommandsPanels, self).__init__(engine, commands)
+
+        if self._panel_commands is None:
+            self._panel_commands = get_registered_panels(engine)
```

Comment:
The `commands` argument for the menu/shelf objects are optional to allow the engine initialization to share those `AppCommand` objects between the classes. This class uses the registered commands as well as the registered panels and follows the same structure, though there are no other classes that share the panels though, so I agree it's not clear why this is necessary. It also makes it possible to add commands/panels that aren't registered (not sure if that makes sense to allow). I will add some additional comments to clarify and go back and document those other classes too. Let me know if you think we should change the code too.